kotlinpoet/src/main/java/com/squareup/kotlinpoet/KmSpecs.kt

```diff
@@ -0,0 +1,405 @@
+      .build()
+}
+
+@KotlinPoetKm
+private fun ImmutableKmProperty.toPropertySpec(
+  typeParamResolver: ((index: Int) -> TypeName),
+  isConstructorParam: Boolean
+) = PropertySpec.builder(name, returnType.toTypeName(typeParamResolver))
+    .apply {
+      addVisibility { addModifiers(it) }
+      addModifiers(*flags.modalities
+          .filterNot { it == KModifier.FINAL && !isOverride }
+          .toTypedArray())
+      if (isOverride) {
+        addModifiers(KModifier.OVERRIDE)
+      }
+      if (isConst) {
+        addModifiers(KModifier.CONST)
+      }
+      if (isVar) {
+        mutable(true)
+      } else if (isVal) {
+        mutable(false)
+      }
+      if (isDelegated) {
+        delegate("") // Placeholder
+      }
+      if (isExpect) {
+        addModifiers(KModifier.EXPECT)
+      }
+      if (isExternal) {
+        addModifiers(KModifier.EXTERNAL)
+      }
+      if (isLateinit) {
+        addModifiers(KModifier.LATEINIT)
+      }
+      if (isConstructorParam || (!isDelegated && !isLateinit)) {
+        // TODO if hasConstant + elements, we could look up the constant initializer
+        when {
+          isConstructorParam -> initializer(name)
+          type.isNullable -> initializer("null")
+          else -> initializer(TODO_BLOCK)
+        }
+      }
+      if (hasGetter) {
+        val visibility = setterFlags.visibility
+        val modalities = setterFlags.modalities
+            .filterNot { it == KModifier.FINAL && !flags.isOverrideProperty }
+        val propertyAccessorFlags = setterFlags.propertyAccessorFlags
+        if (visibility != KModifier.PUBLIC || modalities.isNotEmpty() || propertyAccessorFlags.isNotEmpty()) {
+          getter(FunSpec.setterBuilder()
+              .apply {
+                addModifiers(visibility)
+                addModifiers(*modalities.toTypedArray())
+                addModifiers(*propertyAccessorFlags.toKModifiersArray())
+              }
+              .build())
+        }
+      }
+      if (hasSetter) {
+        val visibility = setterFlags.visibility
+        val modalities = setterFlags.modalities
+            .filterNot { it == KModifier.FINAL && !flags.isOverrideProperty }
+        val propertyAccessorFlags = setterFlags.propertyAccessorFlags
+        if (visibility != KModifier.PUBLIC || modalities.isNotEmpty() || propertyAccessorFlags.isNotEmpty()) {
+          setter(FunSpec.setterBuilder()
+              .apply {
+                addModifiers(visibility)
+                addModifiers(*modalities.toTypedArray())
+                addModifiers(*propertyAccessorFlags.toKModifiersArray())
+              }
+              .build())
+        }
+      }
+    }
+    .tag(this)
+    .build()
+
+private fun Set<PropertyAccessorFlag>.toKModifiersArray(): Array<KModifier> {
+  return map {
+    when (it) {
+      IS_EXTERNAL -> KModifier.EXTERNAL
+      IS_INLINE -> KModifier.INLINE
+      IS_NOT_DEFAULT -> TODO("Wat")
+    }
+  }.toTypedArray()
+}
+
+@KotlinPoetKm
```

Comment:
Btw, why do we care about marking private methods with experimental annotations?

Reply:
I think it's just the nature of how these experimental annotations work, you have to apply them everywhere. Or did you mean why use this vs `@UseExperimental`?

Reply:
No, using our own is the right way to go. I would expect them only be necessary for API, not for the private functions, but if that's how it works I'm fine. No big issue.